lib/queue.js

```diff
@@ -946,139 +939,127 @@ Queue.prototype.getJob = function(jobId){
   return Job.fromId(this, jobId);
 };
 
-// Job counts by type
-// Queue#getJobCountByTypes('completed') => completed count
-// Queue#getJobCountByTypes('completed,failed') => completed + failed count
-// Queue#getJobCountByTypes('completed', 'failed') => completed + failed count
-// Queue#getJobCountByTypes('completed,waiting', 'failed') => completed + waiting + failed count
-Queue.prototype.getJobCountByTypes = function() {
+Queue.prototype._commandByType = function(types, count, callback) {
   var _this = this;
-  var args = _.compact(Array.prototype.slice.call(arguments));
-  var types = _.compact(args.join(',').replace(/ /g, '').split(','));
 
-  var multi = this.multi();
+  return _.map(types, function(type) {
+    type = type === 'waiting' ? 'wait' : type; // alias
 
-  _.each(types, function(type) {
     var key = _this.toKey(type);
+
     switch(type) {
       case 'completed':
       case 'failed':
       case 'delayed':
-        multi.zcard(key);
-        break;
+        return callback(key, count ? 'zcard' : 'zrange');
       case 'active':
       case 'wait':
       case 'paused':
-        multi.llen(key);
-        break;
+        return callback(key, count ? 'llen' : 'lrange');
     }
   });
+};
 
-  return multi.exec().then(function(res){
-    return res.map(function(v) {
-      return v[1];
-    }).reduce(function(a, b) {
-      return a + b;
-    });
-  }) || 0;
+// Job counts by type
+// Queue#getJobCountByTypes('completed') => completed count
+// Queue#getJobCountByTypes('completed,failed') => completed + failed count
+// Queue#getJobCountByTypes('completed', 'failed') => completed + failed count
+// Queue#getJobCountByTypes('completed,waiting', 'failed') => completed + waiting + failed count
+Queue.prototype.getJobCountByTypes = function() {
+  return this.getJobCounts.apply(this, arguments).then(function(result){
+    return _.chain(result).values().sum().value();
+  });
 };
```

Comment:
@manast sorry for the late response. This doesn't actually fetch all jobs, it just sums the values from the object response of `getJobCounts()`. For example, `{ waiting: 3, active: 5 } -> 8`.